cli/cmd/proxy/read/command_test.go

```diff
@@ -35,30 +38,96 @@ func TestArgumentParsing(t *testing.T) {
 
 	for name, tc := range cases {
 		t.Run(name, func(t *testing.T) {
-			c := getInitializedCommand(t)
+			c, _ := getInitializedCommand(t)
 			c.kubernetes = fake.NewSimpleClientset()
 			out := c.Run(tc.args)
 			require.Equal(t, tc.out, out)
 		})
 	}
 }
 
+func TestEndToEnd(t *testing.T) {
+	// These regular expressions must be present in the output.
+	expected := []string{
+		"Envoy configuration for fakePodName in Namespace default:",
+
+		"==> Clusters \\(6\\)",
+		"Name.*FQDN.*Endpoints.*Type.*Last Updated",
+		"local_agent.*local_agent.*192\\.168\\.79\\.187:8502.*STATIC.*2022-05-13T04:22:39\\.553Z",
+		"local_app.*local_app.*127\\.0\\.0\\.1:8080.*STATIC.*2022-05-13T04:22:39\\.655Z",
+		"client.*client\\.default\\.dc1\\.internal\\.bc3815c2-1a0f-f3ff-a2e9-20d791f08d00\\.consul.*EDS.*2022-06-09T00:39:12\\.948Z",
+		"frontend.*frontend\\.default\\.dc1\\.internal\\.bc3815c2-1a0f-f3ff-a2e9-20d791f08d00\\.consul.*EDS.*2022-06-09T00:39:12\\.855Z",
+		"original-destination.*original-destination.*ORIGINAL_DST.*2022-05-13T04:22:39.743Z",
+		"server.*server.default.dc1.internal.bc3815c2-1a0f-f3ff-a2e9-20d791f08d00.consul.*EDS.*2022-06-09T00:39:12\\.754Z",
+
+		"==> Endpoints \\(9\\)",
+		"Address:Port.*Cluster.*Weight.*Status",
+		"192.168.79.187:8502.*local_agent.*1.00.*HEALTHY",
+		"127.0.0.1:8080.*local_app.*1.00.*HEALTHY",
+		"192.168.31.201:20000.*1.00.*HEALTHY",
+		"192.168.47.235:20000.*1.00.*HEALTHY",
+		"192.168.71.254:20000.*1.00.*HEALTHY",
+		"192.168.63.120:20000.*1.00.*HEALTHY",
+		"192.168.18.110:20000.*1.00.*HEALTHY",
+		"192.168.52.101:20000.*1.00.*HEALTHY",
+		"192.168.65.131:20000.*1.00.*HEALTHY",
+
+		"==> Listeners \\(2\\)",
+		"Name.*Address:Port.*Direction.*Filter Chain Match.*Filters.*Last Updated",
+		"public_listener.*192\\.168\\.69\\.179:20000.*INBOUND.*Any.*\\* -> local_app/.*2022-06-09T00:39:27\\.668Z",
+		"outbound_listener.*127.0.0.1:15001.*OUTBOUND.*10\\.100\\.134\\.173/32, 240\\.0\\.0\\.3/32.*-> client.default.dc1.internal.bc3815c2-1a0f-f3ff-a2e9-20d791f08d00.consul.*2022-05-24T17:41:59\\.079Z",
+		"10\\.100\\.254\\.176/32, 240\\.0\\.0\\.4/32.*\\* -> server\\.default\\.dc1\\.internal\\.bc3815c2-1a0f-f3ff-a2e9-20d791f08d00\\.consul/",
+		"10\\.100\\.31\\.2/32, 240\\.0\\.0\\.2/32.*-> frontend\\.default\\.dc1\\.internal\\.bc3815c2-1a0f-f3ff-a2e9-20d791f08d00\\.consul",
+		"Any.*-> original-destination",
+
+		"==> Routes \\(2\\)",
+		"Name.*Destination Cluster.*Last Updated",
+		"public_listener.*local_app/.*2022-06-09T00:39:27.667Z",
+		"server.*server\\.default\\.dc1\\.internal\\.bc3815c2-1a0f-f3ff-a2e9-20d791f08d00\\.consul/.*2022-05-24T17:41:59\\.078Z",
+
+		"==> Secrets \\(2\\)",
+		"Name.*Type.*Last Updated",
+		"default.*Dynamic Active.*2022-05-24T17:41:59.078Z",
+		"ROOTCA.*Dynamic Warming.*2022-03-15T05:14:22.868Z",
+	}
+
+	// A fetchConfig function that just returns the test Envoy config.
+	mockFetchConfig := func(context.Context, common.PortForwarder) (*EnvoyConfig, error) {
```

Comment:
What is the difference between a mock and a fake?

Reply:
Good question. I actually think this is more a "fake" than a "mock" after doing some research. 

There doesn't seem to be any formal definition, but there is some consensus that:
- a mock represents an external _something_ like an API or DB that is hit by the code and tracks that it is reached. 
- a fake is a stand in for some part of the code that obviates some external call, long running process, or something you just don't want running in the test.

I think this is actually a fake.